tests/utils/utils.go

```diff
@@ -48,26 +54,142 @@ var (
+	// Wait for the transaction to be accepted
+	receipt, err := bind.WaitMined(ctx, subnetInfo.ChainRPCClient, tx)
 	Expect(err).Should(BeNil())
-	defer subA.Unsubscribe()
+	if expectSuccess {
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+	} else {
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusFailed))
+	}
+
+	return receipt
+}
+
+func SendCrossChainMessageAndWaitForAcceptance(
+	ctx context.Context,
+	source SubnetTestInfo,
+	destination SubnetTestInfo,
+	input teleportermessenger.TeleporterMessageInput,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	transactor *teleportermessenger.TeleporterMessenger,
+) (*types.Receipt, *big.Int) {
+	opts := CreateTransactorOpts(ctx, source, fundedAddress, fundedKey)
 
-	err = wsClient.SendTransaction(ctx, tx)
+	// Send a transaction to the Teleporter contract
+	txn, err := transactor.SendCrossChainMessage(opts, input)
 	Expect(err).Should(BeNil())
+
 	// Wait for the transaction to be accepted
-	<-newHeads
+	receipt, err := bind.WaitMined(ctx, source.ChainRPCClient, txn)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	// Check the transaction logs for the SendCrossChainMessage event emitted by the Teleporter contract
+	event, err := GetEventFromLogs(receipt.Logs, transactor.ParseSendCrossChainMessage)
+	Expect(err).Should(BeNil())
+	Expect(event.DestinationChainID[:]).Should(Equal(destination.BlockchainID[:]))
+
+	log.Info("Sending SendCrossChainMessage transaction on source chain",
+		"sourceChainID", source.BlockchainID,
+		"destinationChainID", destination.BlockchainID,
+		"txHash", txn.Hash())
+
+	return receipt, event.Message.MessageID
+}
+
+func SendAddFeeAmountAndWaitForAcceptance(
+	ctx context.Context,
+	source SubnetTestInfo,
+	destination SubnetTestInfo,
+	messageID *big.Int,
+	amount *big.Int,
+	feeContractAddress common.Address,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	transactor *teleportermessenger.TeleporterMessenger,
+) *types.Receipt {
+	opts := CreateTransactorOpts(ctx, source, fundedAddress, fundedKey)
+
+	txn, err := transactor.AddFeeAmount(opts, destination.BlockchainID, messageID, feeContractAddress, amount)
+	Expect(err).Should(BeNil())
+	receipt, err := bind.WaitMined(ctx, source.ChainRPCClient, txn)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	addFeeAmountEvent, err := GetEventFromLogs(receipt.Logs, transactor.ParseAddFeeAmount)
+	Expect(err).Should(BeNil())
+	Expect(addFeeAmountEvent.MessageID).Should(Equal(messageID))
+	Expect(addFeeAmountEvent.DestinationChainID[:]).Should(Equal(destination.BlockchainID[:]))
+
+	log.Info("Send AddFeeAmount transaction on source chain",
+		"messageID", messageID,
+		"sourceChainID", source.BlockchainID,
+		"destinationChainID", destination.BlockchainID)
+
+	return receipt
+}
+
+func RetryMessageExecutionAndWaitForAcceptance(
```

Comment:
shouldn't this function take into some form of `RequireGasLimit` to make sure the transaction gets sent with gas > requiredGasLimit to pass?

Reply:
The transactor already estimates the tx gas and sets that as the gas limit. I don't think checking that enough gas is provided is necessary.